From 7fb619f2dd2f4c03e4f16928e79c990f036637e4 Mon Sep 17 00:00:00 2001
From: Gregor Mi <codestruct@posteo.org>
Date: Sun, 11 Nov 2018 12:00:29 +0100
Subject: [PATCH] Terminal plugin: add default shortcut F4 for Toggle Focus

Summary:
(Toggle Focus switches the focus between the editor component
and the Terminal Tool View.)
F4 was chosen because it is consistent with Dolphin (where F4
opens and closes the Terminal Panel) and there is no conflict with
other shortcuts.

{F6388724}

Reviewers: #kate, cullmann

Reviewed By: #kate, cullmann

Subscribers: cullmann, sars, kwrite-devel

Tags: #kate

Differential Revision: https://phabricator.kde.org/D16652
---
 addons/konsole/kateconsole.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/addons/konsole/kateconsole.cpp b/addons/konsole/kateconsole.cpp
index b77fd7427..b227cbd7f 100644
--- a/addons/konsole/kateconsole.cpp
+++ b/addons/konsole/kateconsole.cpp
@@ -149,6 +149,7 @@ KateConsole::KateConsole (KateKonsolePlugin* plugin, KTextEditor::MainWindow *mw
   a = actionCollection()->addAction(QStringLiteral("katekonsole_tools_toggle_focus"));
   a->setIcon(QIcon::fromTheme(QStringLiteral("utilities-terminal")));
   a->setText(i18nc("@action", "&Focus Terminal"));
+  a->setShortcut(QKeySequence(Qt::Key_F4));
   connect(a, &QAction::triggered, this, &KateConsole::slotToggleFocus);

   m_mw->guiFactory()->addClient (this);
